includes/DataDumpPager.php

```diff
@@ -2,185 +2,201 @@
 						]
 					);
 					$token = Html::element(
 						'input',
 						[
 							'type' => 'hidden',
 							'name' => 'token',
-							'value' => $this->getContext()->getCsrfTokenSet()->getToken()
+							'value' => $this->getContext()->getCsrfTokenSet()->getToken(),
 						]
 					);
-					// Do not show a delete button if the dump is not completed or failed.
-					if ( $row->dumps_status === 'completed' || $row->dumps_status === 'failed' ) {
+					// Do not show a delete button if the dump is not completed, failed,
+					// or queued or in progress for over 48 hours.
+					if (
+						$row->dumps_status === 'completed' ||
+						$row->dumps_status === 'failed' ||
+						(
+							(
+								$row->dumps_status === 'queued' ||
+								$row->dumps_status === 'in-progress'
+							) &&
+							( strtotime( $row->dumps_timestamp ) <= time() - 48 * 3600 )
+						)
+					) {
 						$formatted = Html::openElement(
 							'form',
 							[
-								'action' => $link,
-								'method' => 'POST'
+								'action' => $pageTitle->getLinkURL( $query ),
+								'method' => 'POST',
 							]
 						) . $element . $token . Html::closeElement( 'form' );
 					}
 				}
 				break;
 			default:
-				$formatted = "Unable to format $name";
-				break;
+				$formatted = $this->escape( "Unable to format {$name}" );
 		}
 
 		return $formatted;
 	}
 
-	public function getQueryInfo() {
+	/**
+	 * Safely HTML-escapes $value
+	 */
+	private function escape( string $value ): string {
+		return htmlspecialchars( $value, ENT_QUOTES, 'UTF-8', false );
+	}
+
+	/** @inheritDoc */
+	public function getQueryInfo(): array {
 		return [
-			'tables' => [ 'data_dump' ],
-			'fields' => [ 'dumps_status', 'dumps_filename', 'dumps_size', 'dumps_timestamp', 'dumps_type' ],
+			'tables' => [
+				'data_dump',
+			],
+			'fields' => [
+				'dumps_status',
+				'dumps_filename',
+				'dumps_size',
+				'dumps_timestamp',
+				'dumps_type',
+			],
 			'conds' => [],
 			'joins_conds' => [],
 		];
 	}
 
-	public function getDefaultSort() {
+	/** @inheritDoc */
+	public function getDefaultSort(): string {
 		return 'dumps_timestamp';
 	}
 
-	public function isFieldSortable( $name ) {
-		if ( $name === 'dumps_delete' || $name === 'dumps_status' ) {
-			return false;
-		} else {
-			return true;
-		}
+	/** @inheritDoc */
+	public function isFieldSortable( $name ): bool {
+		return $name !== 'dumps_delete' && $name !== 'dumps_status';
 	}
 
-	public function getForm() {
-		$dataDumpDisableGenerate = $this->config->get( 'DataDumpDisableGenerate' );
+	public function getForm(): void {
+		$dataDumpDisableGenerate = $this->config->get( ConfigNames::DisableGenerate );
 		if ( $dataDumpDisableGenerate ) {
 			$out = $this->getOutput();
 			$out->addHTML(
 				Html::errorBox( $this->msg( 'datadump-generated-disabled' )->escaped() )
 			);
 
 			$out->addHTML(
-				'<br />' . Linker::specialLink( 'DataDump', 'datadump-refresh' )
+				Html::closeElement( 'br' ) .
```

Comment:
```suggestion
				Html::element( 'br' ) .
```

Reply:
MediaWiki checks for that, and doesn't write the closing element if so:
https://gerrit.wikimedia.org/r/plugins/gitiles/mediawiki/core/+/1a2ac9f421b8a6d9a5717024772015a39fa8216f/includes/Html/Html.php#193

Reply:
Oh interesting I never knew that lol